<a href="https://colab.research.google.com/github/Kritdanai044/translate/blob/main/translate.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!pip install gdown
!apt-get update                                                                          # อัพเดท Package ทั้งหมดใน VM ตัวนี้


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Get:1 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Hit:2 http://archive.ubuntu.com/ubuntu bionic InRelease
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Hit:4 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease
Ign:5 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease
Get:6 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease [1,581 B]
Get:7 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [83.3 kB]
Hit:8 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release
Get:9 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease [3,626 B]
Hit:10 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease
Hit:11 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic InRelease
Get:12 ht

In [2]:
!pip install translators --upgrade


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 62 kB 1.4 MB/s 
     |████████████████████████████████| 76 kB 5.3 MB/s 
     |████████████████████████████████| 58 kB 6.3 MB/s 
     |████████████████████████████████| 115 kB 55.4 MB/s 
     |████████████████████████████████| 52 kB 1.1 MB/s 
  Created wheel for PyExecJS: filename=PyExecJS-1.5.1-py3-none-any.whl size=14598 sha256=78713c1d703ed503f45ad40fa125ce16d2727ea6d74de086f4b3bd39da53f258
  Stored in directory: /root/.cache/pip/wheels/9a/ee/03/da5c0b4a8c13362beeb844eb913bbe58a89bde1de2b9157007
Successfully built PyExecJS
  Attempting uninstall: requests
    Found existing installation: requests 2.23.0
    Uninstalling requests-2.23.0:
      Successfully uninstalled requests-2.23.0


In [3]:
import translators as ts
import gdown
import pandas as pd

Using state South Carolina server backend.


In [4]:
phrase = 'The quick brown fox jumps over the lazy dog.'
ts.google(phrase, from_language='en', to_language='th')

'สุนัขจิ้งจอกสีน้ำตาลเร็วกระโดดข้ามสุนัขขี้เกียจ'

In [5]:
!gdown https://drive.google.com/uc?id=1FSkLW5ixIrjJZkVTpbftPOeN_2bzRA7y

Downloading...
From: https://drive.google.com/uc?id=1FSkLW5ixIrjJZkVTpbftPOeN_2bzRA7y
To: /content/translate.zip
100% 219M/219M [00:02<00:00, 93.8MB/s]


In [6]:
!unzip translate.zip

Archive:  translate.zip
  inflating: lan_to_language.json    
  inflating: sentences.csv           


In [7]:
# df_l2l = pd.read_json("/content/lan_to_language.json", )
df_sen = pd.read_csv("/content/sentences.csv")

In [8]:
import json

In [9]:
f = open('/content/lan_to_language.json')

In [10]:
data = json.load(f)

In [11]:
data

{'cmn': 'Mandarin Chinese',
 'deu': 'German',
 'rus': 'Russian',
 'fra': 'French',
 'eng': 'English',
 'jpn': 'Japanese',
 'spa': 'Spanish',
 'ita': 'Italian',
 'kor': 'Korean',
 'vie': 'Vietnamese',
 'nld': 'Dutch',
 'epo': 'Esperanto',
 'por': 'Portuguese',
 'tur': 'Turkish',
 'heb': 'Hebrew',
 'hun': 'Hungarian',
 'ell': 'Modern Greek (1453-)',
 'ind': 'Indonesian',
 'ara': 'Arabic',
 'arz': 'Egyptian Arabic',
 'fin': 'Finnish',
 'bul': 'Bulgarian',
 'yue': 'Yue Chinese',
 'swe': 'Swedish',
 'ukr': 'Ukrainian',
 'bel': 'Belarusian',
 'que': 'Quechua',
 'ces': 'Czech',
 'swh': 'Swahili (individual language)',
 'nno': 'Norwegian Nynorsk',
 'wuu': 'Wu Chinese',
 'nob': 'Norwegian Bokmål',
 'zsm': 'Standard Malay',
 'est': 'Estonian',
 'kat': 'Georgian',
 'pol': 'Polish',
 'lat': 'Latin',
 'urd': 'Urdu',
 'sqi': 'Albanian',
 'isl': 'Icelandic',
 'fry': 'Western Frisian',
 'afr': 'Afrikaans',
 'ron': 'Romanian',
 'fao': 'Faroese',
 'san': 'Sanskrit',
 'bre': 'Breton',
 'tat': 'Tatar',
 '

In [12]:
type(data)

dict

In [13]:
l2l = {}

In [14]:
l2l["country"] = data

In [15]:
l2l

{'country': {'cmn': 'Mandarin Chinese',
  'deu': 'German',
  'rus': 'Russian',
  'fra': 'French',
  'eng': 'English',
  'jpn': 'Japanese',
  'spa': 'Spanish',
  'ita': 'Italian',
  'kor': 'Korean',
  'vie': 'Vietnamese',
  'nld': 'Dutch',
  'epo': 'Esperanto',
  'por': 'Portuguese',
  'tur': 'Turkish',
  'heb': 'Hebrew',
  'hun': 'Hungarian',
  'ell': 'Modern Greek (1453-)',
  'ind': 'Indonesian',
  'ara': 'Arabic',
  'arz': 'Egyptian Arabic',
  'fin': 'Finnish',
  'bul': 'Bulgarian',
  'yue': 'Yue Chinese',
  'swe': 'Swedish',
  'ukr': 'Ukrainian',
  'bel': 'Belarusian',
  'que': 'Quechua',
  'ces': 'Czech',
  'swh': 'Swahili (individual language)',
  'nno': 'Norwegian Nynorsk',
  'wuu': 'Wu Chinese',
  'nob': 'Norwegian Bokmål',
  'zsm': 'Standard Malay',
  'est': 'Estonian',
  'kat': 'Georgian',
  'pol': 'Polish',
  'lat': 'Latin',
  'urd': 'Urdu',
  'sqi': 'Albanian',
  'isl': 'Icelandic',
  'fry': 'Western Frisian',
  'afr': 'Afrikaans',
  'ron': 'Romanian',
  'fao': 'Faroese',
  

In [16]:
df_l2l = pd.DataFrame(l2l)

In [17]:
df_l2l

country
abk                    Abkhazian
acm          Mesopotamian Arabic
ady                       Adyghe
afb                  Gulf Arabic
afh                     Afrihili
..                           ...
zgh  Standard Moroccan Tamazight
zlm  Malay (individual language)
zsm               Standard Malay
zul                         Zulu
zza                         Zaza

[404 rows x 1 columns]

In [18]:
df_sen

id lan_code                      sentence
0                1      cmn                        我們試試看！
1                2      cmn                       我该去睡觉了。
2                3      cmn                       你在干什麼啊？
3                4      cmn                        這是什麼啊？
4                5      cmn        今天是６月１８号，也是Muiriel的生日！
...            ...      ...                           ...
10341807  10794524      spa  Quiero este libro por favor.
10341808  10794525      spa           Los han hecho huir.
10341809  10794526      spa                  Los botaron.
10341810  10794527      spa          Los hicieron correr.
10341811  10794528      spa                Los corrieron.

[10341812 rows x 3 columns]

In [19]:
phrase = '我們試試看!'
ts.google(phrase, from_language='zh-CN', to_language='th')

'มาลองดูกัน!'

In [20]:
df_sen["lan_code"].unique()

array(['cmn', 'deu', 'rus', 'fra', 'eng', 'jpn', 'spa', 'ita', 'kor',
       'vie', 'nld', 'epo', 'por', 'tur', 'heb', 'hun', 'ell', 'ind',
       'ara', 'arz', 'fin', 'bul', 'yue', 'swe', 'ukr', 'bel', 'que',
       'ces', 'swh', 'nno', 'wuu', 'nob', 'zsm', 'est', 'kat', 'pol',
       'lat', 'urd', 'sqi', 'isl', 'fry', 'afr', 'ron', 'fao', 'san',
       'bre', 'tat', 'yid', 'uig', 'uzb', 'srp', 'qya', 'dan', 'pes',
       'slk', 'eus', 'cycl', 'acm', 'tgl', 'lvs', 'kaz', 'hye', 'hin',
       'lit', 'ben', 'cat', 'bos', 'hrv', 'tha', 'orv', 'cha', 'mon',
       'lzh', 'scn', 'gle', 'mkd', 'slv', 'frm', 'glg', 'vol', 'ain',
       'jbo', 'tok', 'ina', 'nds', 'mal', 'tlh', 'roh', 'ltz', 'oss',
       'ido', 'gla', 'mlt', 'sco', 'ast', 'jav', 'oci', 'ile', 'ota',
       'xal', 'tel', 'sjn', 'nov', 'khm', 'tpi', 'ang', 'aze', 'tgk',
       'tuk', 'chv', 'hsb', 'dsb', 'bod', 'sme', 'cym', 'mri', 'ksh',
       'kmr', 'ewe', 'kab', 'ber', 'tpw', 'udm', 'lld', 'pms', 'lad',
       'grn', 'mlg'

In [21]:
!apt-get update                                                                          # อัพเดท Package ทั้งหมดใน VM ตัวนี้
!apt-get install openjdk-8-jdk-headless -qq > /dev/null                                  # ติดตั้ง Java Development Kit (จำเป็นสำหรับการติดตั้ง Spark)
!wget -q https://archive.apache.org/dist/spark/spark-3.1.2/spark-3.1.2-bin-hadoop2.7.tgz # ติดตั้ง Spark 3.1.2
!tar xzvf spark-3.1.2-bin-hadoop2.7.tgz                                                  # Unzip ไฟล์ Spark 3.1.2
!pip install -q findspark==1.3.0                                                         # ติดตั้ง Package Python สำหรับเชื่อมต่อกับ Spark 

Hit:1 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:2 http://security.ubuntu.com/ubuntu bionic-security InRelease
Hit:3 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:4 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease
Hit:5 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Ign:6 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease
Hit:7 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease
Hit:8 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release
Hit:9 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease
Hit:10 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease
Hit:11 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic InRelease
Hit:12 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease
Reading package lists... Done
spark-3.1.2-bin-hadoop2.7/
spark-3.1.2-bin-hadoop2.

In [22]:
# Set enviroment variable ให้ Python รู้จัก Spark
import os
os.environ["JAVA_HOME"] = "/usr/lib/jvm/java-8-openjdk-amd64"
os.environ["SPARK_HOME"] = "/content/spark-3.1.2-bin-hadoop2.7"

In [23]:
# ติดตั้ง PySpark ลงใน Python
!pip install pyspark==3.1.2

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 212.4 MB 67 kB/s 
     |████████████████████████████████| 198 kB 5.8 MB/s 
  Created wheel for pyspark: filename=pyspark-3.1.2-py2.py3-none-any.whl size=212880769 sha256=857f11298ea90e199f167f69c50902e79b3b7c6d5ddbd7404cb4bff06a8f4a54
  Stored in directory: /root/.cache/pip/wheels/a5/0a/c1/9561f6fecb759579a7d863dcd846daaa95f598744e71b02c77
Successfully built pyspark


In [24]:
# สร้าง Spark Session เพิ้อใช้งาน Spark
from pyspark.sql import SparkSession
spark = SparkSession.builder.master("local[*]").getOrCreate()

In [25]:
dt = spark.read.csv('/content/sentences.csv', header = True, inferSchema = True, )

In [26]:
dt.show()

+---+--------+--------------------------------------+
| id|lan_code|                              sentence|
+---+--------+--------------------------------------+
|  1|     cmn|                          我們試試看！|
|  2|     cmn|                        我该去睡觉了。|
|  3|     cmn|                        你在干什麼啊？|
|  4|     cmn|                          這是什麼啊？|
|  5|     cmn|       今天是６月１８号，也是Muirie...|
|  6|     cmn|                   生日快乐，Muiriel！|
|  7|     cmn|                   Muiriel现在20岁了。|
|  8|     cmn|                 "密码是""Muiriel""。"|
|  9|     cmn|                      我很快就會回來。|
| 10|     cmn|                            我不知道。|
| 11|     cmn|              我不知道應該說什麼才好。|
| 12|     cmn|                    這個永遠完不了了。|
| 13|     cmn|          我只是不知道應該說什麼而已……|
| 14|     cmn|                那是一隻有惡意的兔子。|
| 15|     cmn|                        我以前在山里。|
| 16|     cmn|                      那是一张近照吗？|
| 17|     cmn|                我不知道我有沒有時間。|
| 18|     cmn|剛才我的麥克風沒起作用，不知道為什麼。|
| 19|     cmn|      到

In [27]:
dt.describe().show()

+-------+-----------------+--------------------+--------------------+
|summary|               id|            lan_code|            sentence|
+-------+-----------------+--------------------+--------------------+
|  count|         10358184|            10344605|            10342389|
|   mean|5456834.004596196|                null|              2018.0|
| stddev|3126110.219602189|                null|                null|
|    min|                1| 'Can you smile f...| ""which would su...|
|    max|          9999999|         ” she said.|           𡀲係書。|
+-------+-----------------+--------------------+--------------------+



In [28]:
dt.filter((dt.lan_code == "cmn") | (dt.lan_code == "jpn") | (dt.lan_code == "kor") | (dt.lan_code == "tha")).describe().show()

+-------+------------------+--------+-----------------------------------+
|summary|                id|lan_code|                           sentence|
+-------+------------------+--------+-----------------------------------+
|  count|            306274|  306274|                             306274|
|   mean|2562487.7549514486|    null|                               null|
| stddev| 3610521.918777409|    null|                               null|
|    min|                 1|     cmn| 我已經長大了不玩蟲了，這個男孩說。|
|    max|             99999|     tha|                       𠮟っとくよ。|
+-------+------------------+--------+-----------------------------------+



In [29]:
dt.filter(dt.lan_code == "kor").describe().show()

+-------+-----------------+--------+-----------------------------+
|summary|               id|lan_code|                     sentence|
+-------+-----------------+--------+-----------------------------+
|  count|             8981|    8981|                         8981|
|   mean|6806475.127045986|    null|                         null|
| stddev|3056695.721349035|    null|                         null|
|    min|         10008440|     kor|" 톰은 좀 어때? 마음을 좀 ...|
|    max|           998746|     kor|                 힘이 정의다.|
+-------+-----------------+--------+-----------------------------+



In [30]:
dt.printSchema()

root
 |-- id: string (nullable = true)
 |-- lan_code: string (nullable = true)
 |-- sentence: string (nullable = true)



In [31]:
# final_dt =  dt.filter((dt.lan_code == "cmn") | (dt.lan_code == "jpn") | (dt.lan_code == "kor") | (dt.lan_code == "tha"))
final_dt =  dt.filter(dt.lan_code == "tha")

In [32]:
final_dt.describe().show()

+-------+------------------+--------+--------------------+
|summary|                id|lan_code|            sentence|
+-------+------------------+--------+--------------------+
|  count|              4751|    4751|                4751|
|   mean| 8376872.311302884|    null|                null|
| stddev|2218164.9307905426|    null|                null|
|    min|          10064258|     tha|"ขอบคุณ ""ไม่เป็น...|
|    max|            983686|     tha|ไวรัสกำลังแพร่กระ...|
+-------+------------------+--------+--------------------+



In [33]:
pdfinal = final_dt.toPandas()

In [34]:
pdfinal.to_csv("output.csv")

In [35]:
pdfinal

id lan_code                                       sentence
0       477173      tha                          คุณแต่งงานแล้วหรือยัง
1       477178      tha               ภาษาไทยเป็นภาษาราชการในประเทศไทย
2       477604      tha                        ผมอ่านเดอะนิวยอร์กไทมส์
3       477650      tha                                       ลองทำดู!
4       478296      tha                              ภาษาเดียวไม่เคยพอ
...        ...      ...                                            ...
4746  10777086      tha  ฉันคิดว่าทอมและแมรีกำลังขายอะไรสักอย่างให้เรา
4747  10777087      tha                              ทอมซื้อขนมปังอยู่
4748  10777088      tha                        ถึงเวลาเริ่มพิธีการแล้ว
4749  10777089      tha                               พวกเขาดูตื่นตาใจ
4750  10789622      tha  ฉันพอแล้วกับความยากจนต่อต้านวิทยาศาสตร์ของคุณ

[4751 rows x 3 columns]

In [36]:
df_l2l.dtypes

country    object
dtype: object

In [37]:
df_l2l = df_l2l.reset_index().rename(columns={"index": "code"})
df_l2l

code                      country
0    abk                    Abkhazian
1    acm          Mesopotamian Arabic
2    ady                       Adyghe
3    afb                  Gulf Arabic
4    afh                     Afrihili
..   ...                          ...
399  zgh  Standard Moroccan Tamazight
400  zlm  Malay (individual language)
401  zsm               Standard Malay
402  zul                         Zulu
403  zza                         Zaza

[404 rows x 2 columns]

In [38]:
final_df = pdfinal.merge(df_l2l, how="left", left_on="lan_code", right_on="code")

In [39]:
final_df

id lan_code                                       sentence code  \
0       477173      tha                          คุณแต่งงานแล้วหรือยัง  tha   
1       477178      tha               ภาษาไทยเป็นภาษาราชการในประเทศไทย  tha   
2       477604      tha                        ผมอ่านเดอะนิวยอร์กไทมส์  tha   
3       477650      tha                                       ลองทำดู!  tha   
4       478296      tha                              ภาษาเดียวไม่เคยพอ  tha   
...        ...      ...                                            ...  ...   
4746  10777086      tha  ฉันคิดว่าทอมและแมรีกำลังขายอะไรสักอย่างให้เรา  tha   
4747  10777087      tha                              ทอมซื้อขนมปังอยู่  tha   
4748  10777088      tha                        ถึงเวลาเริ่มพิธีการแล้ว  tha   
4749  10777089      tha                               พวกเขาดูตื่นตาใจ  tha   
4750  10789622      tha  ฉันพอแล้วกับความยากจนต่อต้านวิทยาศาสตร์ของคุณ  tha   

     country  
0       Thai  
1       Thai  
2       Thai  
3       Thai  
4       Thai  
...      ...  
4746    Thai  
4747    Thai  
4748    Thai  
4749    Thai  
4750    Thai  

[4751 rows x 5 columns]

In [40]:
final_df = final_df.drop("code", axis=1)

In [41]:
final_df

id lan_code                                       sentence country
0       477173      tha                          คุณแต่งงานแล้วหรือยัง    Thai
1       477178      tha               ภาษาไทยเป็นภาษาราชการในประเทศไทย    Thai
2       477604      tha                        ผมอ่านเดอะนิวยอร์กไทมส์    Thai
3       477650      tha                                       ลองทำดู!    Thai
4       478296      tha                              ภาษาเดียวไม่เคยพอ    Thai
...        ...      ...                                            ...     ...
4746  10777086      tha  ฉันคิดว่าทอมและแมรีกำลังขายอะไรสักอย่างให้เรา    Thai
4747  10777087      tha                              ทอมซื้อขนมปังอยู่    Thai
4748  10777088      tha                        ถึงเวลาเริ่มพิธีการแล้ว    Thai
4749  10777089      tha                               พวกเขาดูตื่นตาใจ    Thai
4750  10789622      tha  ฉันพอแล้วกับความยากจนต่อต้านวิทยาศาสตร์ของคุณ    Thai

[4751 rows x 4 columns]

In [42]:
final_df["tran_code"] = final_df["lan_code"]

cmn = zh-CN
jpn = ja
kor = ko
tha = th

 **ยิ่งข้อมูลเยอะยิ่งใช้เวลาในการแปลนานขึ้น ในนี้เลยทดลองแปลแค่ภาษาไทย
(4751 แถว ใช้เวลาประมาณ 1.30 ชั่วโมง)**

In [43]:
# converter = lambda x : x*2 if x < 10 else (x*3 if x < 20 else x)

# final_df["tran_code"] = final_df.apply(lambda x: x["lan_code"].replace("cmn", "zh-CN") if x["lan_code"] == "cmn" else (x["lan_code"].replace("jpn", "jp") if x["lan_code"] == "jpn") else (x["lan_code"].replace("kor", "ko") if x["lan_code"] == "kor") else (x["lan_code"].replace("tha", "th") if x["lan_code"] == "tha") else None)
# final_df["tran_code"] = final_df.apply(lambda x: x["tran_code"].replace("cmn", "zh-CN"), axis=1)

# final_df["tran_code"] = final_df.apply(lambda x: x["tran_code"].replace("jpn", "jp"), axis=1)
# final_df["tran_code"] = final_df.apply(lambda x: x["tran_code"].replace("kor", "ko"), axis=1)
final_df["tran_code"] = final_df.apply(lambda x: x["tran_code"].replace("tha", "th"), axis=1)

In [44]:
final_df

id lan_code                                       sentence  \
0       477173      tha                          คุณแต่งงานแล้วหรือยัง   
1       477178      tha               ภาษาไทยเป็นภาษาราชการในประเทศไทย   
2       477604      tha                        ผมอ่านเดอะนิวยอร์กไทมส์   
3       477650      tha                                       ลองทำดู!   
4       478296      tha                              ภาษาเดียวไม่เคยพอ   
...        ...      ...                                            ...   
4746  10777086      tha  ฉันคิดว่าทอมและแมรีกำลังขายอะไรสักอย่างให้เรา   
4747  10777087      tha                              ทอมซื้อขนมปังอยู่   
4748  10777088      tha                        ถึงเวลาเริ่มพิธีการแล้ว   
4749  10777089      tha                               พวกเขาดูตื่นตาใจ   
4750  10789622      tha  ฉันพอแล้วกับความยากจนต่อต้านวิทยาศาสตร์ของคุณ   

     country tran_code  
0       Thai        th  
1       Thai        th  
2       Thai        th  
3       Thai        th  
4       Thai        th  
...      ...       ...  
4746    Thai        th  
4747    Thai        th  
4748    Thai        th  
4749    Thai        th  
4750    Thai        th  

[4751 rows x 5 columns]

In [45]:
# phrase = 'ฉันพอแล้วกับความยากจนต่อต้านวิทยาศาสตร์ของคุณ		'
# ts.google(phrase, from_language='th', to_language='en')

'I am enough with the poverty against your science.'

In [46]:
final_df["tran_code"].unique()

array(['th'], dtype=object)

In [47]:
final_df['to_Eng'] = final_df.apply(lambda x: ts.google(x["sentence"], from_language=x["tran_code"], to_language='en') ,axis=1)

In [48]:
final_df

id lan_code                                       sentence  \
0       477173      tha                          คุณแต่งงานแล้วหรือยัง   
1       477178      tha               ภาษาไทยเป็นภาษาราชการในประเทศไทย   
2       477604      tha                        ผมอ่านเดอะนิวยอร์กไทมส์   
3       477650      tha                                       ลองทำดู!   
4       478296      tha                              ภาษาเดียวไม่เคยพอ   
...        ...      ...                                            ...   
4746  10777086      tha  ฉันคิดว่าทอมและแมรีกำลังขายอะไรสักอย่างให้เรา   
4747  10777087      tha                              ทอมซื้อขนมปังอยู่   
4748  10777088      tha                        ถึงเวลาเริ่มพิธีการแล้ว   
4749  10777089      tha                               พวกเขาดูตื่นตาใจ   
4750  10789622      tha  ฉันพอแล้วกับความยากจนต่อต้านวิทยาศาสตร์ของคุณ   

     country tran_code                                             to_Eng  
0       Thai        th                                   Are you married?  
1       Thai        th           Thai is a official language in Thailand.  
2       Thai        th                         I read The New York Times.  
3       Thai        th                                      Try to do it!  
4       Thai        th                      One language is never enough.  
...      ...       ...                                                ...  
4746    Thai        th  I think Tom and Mary are selling something to us.  
4747    Thai        th                                   Tom bought bread  
4748    Thai        th                   It's time to start the ceremony.  
4749    Thai        th                           They seem to be dazzled.  
4750    Thai        th  I am enough with the poverty against your scie...  

[4751 rows x 6 columns]

In [49]:
final_df.to_csv("To_Eng_output.csv")